src/H5.c

```diff
@@ -1124,6 +1198,34 @@ H5is_library_threadsafe(hbool_t *is_ts)
     FUNC_LEAVE_API_NOINIT(SUCCEED)
 } /* end H5is_library_threadsafe() */
 
+/*-------------------------------------------------------------------------
+ * Function:	H5is_library_terminating
+ *
+ * Purpose:	Checks to see if the library is shutting down.
+ *
+ * Note:	Useful for plugins to detect when the library is terminating.
+ *		For example, a VOL connector could check if a "file close"
+ *		callback was the result of the library shutdown process, or
+ *		an API action from the application.
+ *
+ * Return:	SUCCEED/FAIL
+ *
+ *-------------------------------------------------------------------------
+ */
+herr_t
+H5is_library_terminating(hbool_t *is_terminating)
+{
+    FUNC_ENTER_API_NOINIT
+    H5TRACE1("e", "*b", is_terminating);
+
+    HDassert(is_terminating);
```

Comment:
Should use real error checking and not assert() in public API calls.

Reply:
That's not possible here, since the library may not be initialized.

Reply:
But you can still check the parameter and jump to the end returning FAIL if the parameter is NULL.

Reply:
OK, sure - I've changed the code to do that for both H5is_library_terminating as well as the existing H5is_library_threadsafe routine.